zap/src/main/java/org/zaproxy/zap/core/scanner/InputVectorBuilder.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2021 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.core.scanner;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.parosproxy.paros.core.scanner.NameValuePair;
+import org.zaproxy.zap.core.scanner.InputVector.PayloadFormat;
+
+/**
+ * {@code InputVectorBuilder} provides an easy way to build {@code InputVectors} and validates the
+ * provided inputs.
+ *
+ * @author preetkaran20@gmail.com KSASAN
+ */
+public class InputVectorBuilder {
+
+    private Map<Integer, InputVector> inputVectorsMap = new HashMap<>();
+
+    public InputVectorBuilder addAppParameter(
+            NameValuePair nameValuePair, String param, String value, PayloadFormat payloadFormat) {
+        inputVectorsMap.put(
+                nameValuePair.getPosition(),
+                new InputVector(nameValuePair.getPosition(), param, value, payloadFormat));
+        return this;
+    }
+
+    public InputVectorBuilder addAppParameterWithoutParam(
+            NameValuePair nameValuePair, String value, PayloadFormat payloadFormat) {
+        return this.addAppParameter(nameValuePair, null, value, payloadFormat);
```

Comment:
`null` has special meaning (e.g. remove the IV), it should use the original values of the `NameValuePair`.